rest/server_context.go

```diff
@@ -946,6 +954,92 @@ func (sc *ServerContext) getDbConfigFromServer(dbName string) (*DbConfig, error)
 	return &config, nil
 }
 
+//////// STATS LOGGING
+
+func (sc *ServerContext) startStatsLogger() {
+
+	statsLogFrequencySecs := 5
+	if sc.config.Unsupported != nil && sc.config.Unsupported.Logging != nil {
+		statsLogFrequencySecs = sc.config.Unsupported.Logging.StatsLogFrequencySecs
+	}
+
+	interval := time.Second * time.Duration(statsLogFrequencySecs)
+
+	sc.statsLoggingTicker = time.NewTicker(interval)
+	go func() {
+		for range sc.statsLoggingTicker.C {
+			sc.logStats()
+		}
+	}()
+	base.Infof(base.KeyAll, "Logging stats with frequency: %v", interval)
+
+}
+
+func (sc *ServerContext) stopStatsLogger() {
+	if sc.statsLoggingTicker != nil {
+		sc.statsLoggingTicker.Stop()
+	}
+}
+
+func (sc *ServerContext) logStats() {
+
+	AddGoRuntimeStats()
+
+	// Create wrapper expvar map in order to add a timestamp field for logging purposes
+	logStats := new(expvar.Map).Init()
+	logStats.Set("stats", base.Stats)
+	logStats.Set("unix_epoch_timestamp", base.ExpvarInt64Val(time.Now().Unix()))
+
+	// Marshal expvar map w/ timestamp to string and write to logs
+	base.Statsf([]byte(logStats.String()))
```

Comment:
If this is the only caller of Statsf, should we change the signature to accept string?

Reply:
Fixed